snowplow-tracker/src/main/java/com/snowplowanalytics/core/tracker/Tracker.kt

```diff
@@ -333,8 +328,9 @@ class Tracker(
         override fun apply(data: Map<String, Any>) {
             if (screenViewAutotracking) {
                 val event = data["event"] as? ScreenView?
-                event?.let { event ->
-                    getScreenState()?.let { state ->
+                if (event != null) {
```

Comment:
It had a slightly different behaviour which I think was unintended – the `?: track(event)` part could be executed even in case the previous `let` block was evaluated but returned `null`. I think this if-else better reflects what should happen and is a bit more clear.